x-pack/plugins/alerting/server/lib/rule_execution_status.ts

```diff
@@ -80,6 +81,15 @@ export function executionStatusFromState({
     };
   }
 
+  // Set warning status if last run reported any warnings and framework has not set any warnings
+  if (warningsFromLastRun.length > 0 && !warning) {
```

Comment:
we currently only have the ability to set one type of warning so if the framework has already set a warning, that takes precedence over any warnings reported by the rule executor.